File: features/step-definitions/todo-steps.js
import { elementsWithContent, firstTextInput, nthElement, pressEnter } from '../support/utils'

module.exports = function () {
  this.When(/^I create some todos$/, function (done) {
    const input = firstTextInput(this.browser)

    this.browser.fill(input, 'First Todo')
    pressEnter(this.browser, input)

    this.browser.fill(input, 'Second Todo')
    pressEnter(this.browser, input)

    done()
  })

  this.Then(/^I should( not)? see a "([^"]+)" todo$/, function (not, name, done) {
    const todos = elementsWithContent(this.browser, '.view label', name)
    const expectedElements = not ? 0 : 1

    if (todos.length !== expectedElements) {
      throw new Error(`Expected to see 1 "${name}" todo but there were ${todos.length}.`)
    }

    done()
  })

  this.When(/^I mark the (\d+)(?:st|nd|rd|th) todo as complete$/, function (nth) {
    return this.browser.pressButton(nthElement(this.browser, nth, 'toggle'))
  })

  this.Then(/^the todo count should say "([^"]+)"$/, function (value) {
    return this.browser.assert.text('.todo-count', value)
  })
}

File: features/step_definitions/todo_steps.rb
Then(/^I should see a "([^"]*)" todo$/) do |value|
  expect(page).to have_css('.view label', text: value)
end

Then(/^I should not see a "([^"]*)" todo$/) do |value|
  expect(page).to have_no_css('.view label', text: value)
end

When(/^I create some todos$/) do
  input = find('.new-todo')

  input.set('First Todo')
  input.native.send_keys(:return)

  input.set('Second Todo')
  input.native.send_keys(:return)
end

Then(/^the todo count should say "([^"]*)"$/) do |value|
  expect(page).to have_css('.todo-count', text: value)
end

When(/^I mark the (\d+)(?:st|nd|rd|th) todo as complete$/) do |nth|
  click_nth_button('toggle', nth)
end
